rfcs/20240311-verbose-logging-support/README.md

```diff
@@ -0,0 +1,87 @@
+
+|   | `spdlog` Level        | oneDNN Verbose Mode                                              |
+|---|-----------------------|------------------------------------------------------------------|
+| 0 | SPDLOG_LEVEL_TRACE    | `all`                                                            |
+| 1 | SPDLOG_LEVEL_DEBUG    | `dispatch`, `debuginfo=<level>`, `check`                         |
+| 2 | SPDLOG_LEVEL_INFO     | `profile_create`, `profile_exec`, `profile`                      |
+| 3 | SPDLOG_LEVEL_WARNING  | ---                                                              |
+| 4 | SPDLOG_LEVEL_ERROR    | `error`                                                          |
+| 5 | SPDLOG_LEVEL_CRITICAL | -                                                                |
+| 6 | SPDLOG_LEVEL_OFF      | `none`                                                           |
+
+With this alignment, the tracing information printed out for each verbose mode can be also logged at the aligned level. Obviously, the logging level here is determined from the value of the `ONEDNN_VERBOSE` variable. 
+To exercise better control over the logged data, it is recommended to introduce another variable `ONEDNN_LOG_VERBOSE` which specifies the verbose mode for the logged data and also additionally allows the user to use the [`filter`]((https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html?highlight=onednn_verbose#run-time-controls)) option for the verbose mode to filter the information being logged for supported components and primitives. 
+
+### 4. Formatting Specifications for Logging
+The printed verbose information when ONEDNN_VERBOSE=all is formatted and contains the following fields as described [here](https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html#decrypting-the-output):
+```
+onednn_verbose,primitive,info,template:timestamp,operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
+onednn_verbose,graph,info,template:timestamp,operation,engine,partition_id,partition_kind,op_names,data_formats,logical_tensors,fpmath_mode,backend,exec_time
+onednn_verbose,1693533460193.346924,primitive,create:cache_miss,cpu,convolution,jit:avx512_core,forward_training,src_f32:a:blocked:aBcd16b::f0 wei_f32:a:blocked:ABcd16b16a::f0 bia_f32:a:blocked:a::f0 dst_f32:a:blocked:aBcd16b::f0,,alg:convolution_direct,mb2_ic16oc16_ih7oh7kh5sh1dh0ph2_iw7ow7kw5sw1dw0pw2,0.709961
+```
+
+When logging is enabled, the data will be recorded in the logfiles with the same format and field albeit with a few updates:
+- ONEDNN_VERBOSE_TIMESTAMP will be enabled for the logger and will be displayed in the logfile in the format `[%Y-%M-%D %H:%M:%S]`.
```

Comment:
This is definitely wouldn't be appropriate for users who requested timestamp feature. The whole point of it was having a straightforward and linear order of values in ms that is easy to work with in csv. The new format is definitely not helpful.

Reply:
There can be two alternative formats that can keep the timestamps:
- print the verbose outputs as they are with the spdlog fields. This will allow one to process the logfiles in csv without losing any data.
```
[2024-07-03 16:18:23.632] [onednn] [info] onednn_verbose,1693533460193.346924,primitive,create:cache_miss,cpu,convolution,jit:avx512_core,forward_training,src_f32:a:blocked:aBcd16b::f0 wei_f32:a:blocked:ABcd16b16a::f0 bia_f32:a:blocked:a::f0 dst_f32:a:blocked:aBcd16b::f0,,alg:convolution_direct,mb2_ic16oc16_ih7oh7kh5sh1dh0ph2_iw7ow7kw5sw1dw0pw2,0.709961
```
- print the verbose outputs while removing all spdlog fields. The printed data will then be identical to what is displayed on the console.
```
onednn_verbose,1693533460193.346924,primitive,create:cache_miss,cpu,convolution,jit:avx512_core,forward_training,src_f32:a:blocked:aBcd16b::f0 wei_f32:a:blocked:ABcd16b16a::f0 bia_f32:a:blocked:a::f0 dst_f32:a:blocked:aBcd16b::f0,,alg:convolution_direct,mb2_ic16oc16_ih7oh7kh5sh1dh0ph2_iw7ow7kw5sw1dw0pw2,0.709961
```

Reply:
First one should be fine.